### DISCUSSION TITLE
Hide a column, but still show its filter

### DISCUSSION THREAD

**Discussion by jessehas** [TanStack/table#3126] [Q&A] [answered]
I have a situation where I want to hide a column from the user, but still make its filter available since it's useful. The column itself only contains a yes/no flag and is a waste of space to actually show in the table. Any help would be appreciated. Thanks!

I'm rendering filters like this:
```javascript
headerGroup.headers.map((column, index) => column.canFilter ? <div key={index}>{column.render('Filter')}</div> : null)
```

**Comment by jessehas** [comment, marked as answer]
Resolved by using `allColumns` rather than `headerGroup`.

**Comment by tychenjiajun** [comment]
make it a hidden column in initialState
